### PR TITLE
SC-9. Feature-1. Save contacts

### PR DESCRIPTION
**Pull request by DayOut**

### CODE REVIEW

File: src/utils/input_parser.py [hunk 1]
```diff
@@ -2,3 +2,9 @@ def parse_input(user_input: str):
     cmd, *args = user_input.split()
     cmd = cmd.strip().lower()
     return cmd, *args
+
+def capitalize_name(func):
+    def inner(*args, **kwargs):
+        args[0][0] = str(args[0][0]).lower().capitalize()
+        return func(*args, **kwargs)
+    return inner
```
Comment (yuratouch) [inline src/utils/input_parser.py]: Capitalize decorator as improvement approved. 
Please create a new file src/utils/capitalizer.py for this decorator. 
Update import routes.

**Review by yuratouch — CHANGES_REQUESTED** [review]
Please update with discussed changes.

File: src/utils/input_parser.py (hunk 1 above)
Comment (DayOut) [reply src/utils/input_parser.py]: Fixed this moment

File: src/modules/email.py [hunk 2]
```diff
@@ -0,0 +1,8 @@
+from src.modules.field import Field
+
+
+# TODO: Improvement. Refactor. Add error handler decorator
+
+class Email(Field):
+    def __init__(self, value):
+        super().__init__(value)
```
Comment (yuratouch) [inline src/modules/email.py]: rename value to email
add class attribute email.

    def __init__(self, email):
        super().__init__(email)
        self.email = Field(email)

File: src/modules/contact.py [hunk 3]
```diff
@@ -35,7 +38,21 @@ def add_birthday(self, birthday):
...
+        except ValueError as e:
+            print(e)
+
     def __str__(self):
         return (f"Contact name: {self.name.value},"
                 f" phones: {'; '.join(p.value for p in self.phones)},"
-                f" birthday: {self.birthday}")
+                f" birthday: {self.birthday}"
+                f" email: {self.email}"
+                f" address: {self.address}")
```
Comment (yuratouch) [inline src/modules/contact.py:58]: This will throw an error if Contact does not have self.email or self.address

File: src/modules/address.py [hunk 4]
```diff
@@ -0,0 +1,8 @@
+from src.modules.field import Field
+
+
+# TODO: Improvement. Refactor. Add error handler decorator
+
+class Address(Field):
+    def __init__(self, value):
+        super().__init__(value)
```
Comment (yuratouch) [inline src/modules/address.py]: add class attribute address.
rename variable value to address

File: src/handlers/contacts_handler.py [hunk 5]
```diff
@@ -59,11 +99,14 @@ def show_all(_: list, book: ContactBook) -> str:
     res = "\n".join(
         f"{record.name.value}{f"({record.birthday.value})" if record.birthday and record.birthday.value else ""}: "
         f"{", ".join(phone.value for phone in record.phones) if record.phones else "The contact has no phone numbers"}"
+        f"{f" | Email: {record.email.value}" if record.email and record.email.value else ""}"
+        f"{f" | Address:  {record.address.value}" if record.address and record.address.value else ""}"
```
Comment (yuratouch) [inline src/handlers/contacts_handler.py:110]: this will throw an error if contact does not have email or address

**Review by yuratouch — CHANGES_REQUESTED** [review]

File: src/handlers/contacts_handler.py (hunk 5 above)
Comment (DayOut) [reply src/handlers/contacts_handler.py:110]: Discussed this case in Slack. It was not error with function, it is problem with local storage file.

File: src/modules/contact.py (hunk 3 above)
Comment (DayOut) [reply src/modules/contact.py:58]: Discussed this case in Slack. It was not error with function, it is problem with local storage file.

**Review by yuratouch — APPROVED** [review]
Approved for merge

**Pull request merged**